htdocs/resource/class/dolresource.class.php

```diff
@@ -618,7 +615,7 @@ public function update_element_resource($user = null, $notrigger = 0)
 
 		// Update request
 		$sql = "UPDATE ".MAIN_DB_PREFIX."element_resources SET";
-		$sql .= " resource_id=".(isset($this->resource_id) ? "'".$this->db->escape($this->resource_id)."'" : "null").",";
+		$sql .= " resource_id=".(isset($this->resource_id) ? $this->resource_id : "null").",";
```

Comment:
To avoid sql injection, we must force the cast to (int) during the sql forging when data is int (even if already cast to int previously).

Reply:
Thanks